iceoryx_hoofs/posix/filesystem/source/file.cpp

```diff
@@ -0,0 +1,438 @@
+                return iox::error<FileCreationError>(FileCreationError::PermissionDenied);
+            }
+        }
+
+        return iox::success<File>(std::move(file));
+    }
+
+    switch (result.get_error().errnum)
+    {
+    case EACCES:
+        IOX_LOG(ERROR) << "Unable to open/create file due to insufficient permissions.";
+        return iox::error<FileCreationError>(FileCreationError::PermissionDenied);
+    case EPERM:
+        IOX_LOG(ERROR) << "Unable to open/create file due to insufficient permissions.";
+        return iox::error<FileCreationError>(FileCreationError::PermissionDenied);
+    case EINTR:
+        IOX_LOG(ERROR) << "Unable to open/create file since an interrupt signal was received.";
+        return iox::error<FileCreationError>(FileCreationError::Interrupt);
+    case EISDIR:
+        IOX_LOG(ERROR) << "Unable to open/create file since it is actually a directory.";
+        return iox::error<FileCreationError>(FileCreationError::IsDirectory);
+    case ELOOP:
+        IOX_LOG(ERROR) << "Unable to open/create file since too many symbolic links were encountered.";
+        return iox::error<FileCreationError>(FileCreationError::TooManySymbolicLinksEncountered);
+    case EMFILE:
+        IOX_LOG(ERROR) << "Unable to open/create file since the process limit of open file descriptors was reached.";
+        return iox::error<FileCreationError>(FileCreationError::ProcessLimitOfOpenFileDescriptorsReached);
+    case ENFILE:
+        IOX_LOG(ERROR) << "Unable to open/create file since the system limit of open file descriptors was reached.";
+        return iox::error<FileCreationError>(FileCreationError::SystemLimitOfOpenFileDescriptorsReached);
+    case ENOENT:
+        if (print_error_on_non_existing_file)
+        {
+            IOX_LOG(ERROR) << "Unable to open file since the file does not exist.";
+        }
+        return iox::error<FileCreationError>(FileCreationError::DoesNotExist);
+    case ENOMEM:
+        IOX_LOG(ERROR) << "Unable to open/create file due to insufficient memory.";
+        return iox::error<FileCreationError>(FileCreationError::InsufficientMemory);
+    case EOVERFLOW:
+        IOX_LOG(ERROR) << "Unable to open/create file since it is too large.";
+        return iox::error<FileCreationError>(FileCreationError::FileTooLarge);
+    case ETXTBSY:
+        IOX_LOG(ERROR) << "Unable to open/create file since it is currently in use.";
+        return iox::error<FileCreationError>(FileCreationError::CurrentlyInUse);
+    case EEXIST:
+        IOX_LOG(ERROR) << "Unable to create file since it already exists.";
+        return iox::error<FileCreationError>(FileCreationError::AlreadyExists);
+    default:
+        IOX_LOG(ERROR) << "Unable to open/create file due to insufficient permissions.";
+        return iox::error<FileCreationError>(FileCreationError::UnknownError);
+    }
+
+    return iox::success<File>(File(result->value, m_access_mode));
```

Comment:
This is dead code. There is either a return in the if-statement or the switch-statement. It would be easier to read if the error logging would be moved to a separate function and do an early return like `if (result.has_error()) { return translateToError(result.get_error().errnum); }`